net/address/address.go

```diff
@@ -91,6 +145,21 @@ func ParseCIDR(s string) (CIDR, error) {
 	}
 }
 
+func NewCIDRs(ranges []Range) (cidrs []CIDR) {
+	for _, r := range ranges {
+		cidrs = append(cidrs, r.CIDRs()...)
+	}
+	return cidrs
+}
+
+func (cidr CIDR) Start() Address {
+	return cidr.Addr
+}
+
+func (cidr CIDR) End() Address {
+	return Add(cidr.Addr, cidr.Size()-1)
```

Comment:
The `End` of a Range is exclusive, so it is confusing to use the same name for an inclusive end.

Since this `End()` is only used to compare against other calls of the same function, it won't matter which you use.

However, I would probably just indirect via `Range()` and get rid of this function.